test/unit/org/apache/cassandra/transport/messages/ResultMessageTest.java

```diff
@@ -75,13 +99,30 @@ public void testSetKeyspace()
     @Test
     public void testRows()
     {
+        FieldIdentifier f1 = field("f1");  // has field position 0
+        FieldIdentifier f2 = field("f2");  // has field position 1
+        UserType udt = new UserType("ks1",
+                                    bb("myType"),
+                                    asList(f1, f2),
+                                    asList(Int32Type.instance, UTF8Type.instance),
+                                    false);
+
         ColumnSpecification cs1 = new ColumnSpecification("ks1", "cf1", new ColumnIdentifier("a", true), Int32Type.instance);
-        ColumnSpecification cs2 = new ColumnSpecification("ks2", "cf2", new ColumnIdentifier("b", true), Int32Type.instance);
-        ResultSet.ResultMetadata resultMetadata = new ResultSet.ResultMetadata(Arrays.asList(cs1, cs2));
+        ColumnSpecification cs2 = new ColumnSpecification("ks1", "cf1", new ColumnIdentifier("b", true), Int32Type.instance);
+        ColumnSpecification cs3 = new ColumnSpecification("ks1", "cf1", new ColumnIdentifier("c", true), udt);
+        ColumnSpecification cs4 = new ColumnSpecification("ks1", "cf1", new ColumnIdentifier("d", true), new TupleType(asList(Int32Type.instance, udt)));
+        ColumnSpecification cs5 = new ColumnSpecification("ks1", "cf1", new ColumnIdentifier("e", true), ReversedType.getInstance(Int32Type.instance));
+        ColumnSpecification cs6 = new ColumnSpecification("ks1", "cf1", new ColumnIdentifier("f", true), MapType.getInstance(Int32Type.instance, udt, false));
+        ColumnSpecification cs7 = new ColumnSpecification("ks1", "cf1", new ColumnIdentifier("g", true), ListType.getInstance(udt, false));
+        ColumnSpecification cs8 = new ColumnSpecification("ks1", "cf1", new ColumnIdentifier("h", true), SetType.getInstance(udt, false));
+        ColumnSpecification cs9 = new ColumnSpecification("ks1", "cf1", new ColumnIdentifier("i", true), CompositeType.getInstance(Int32Type.instance, UTF8Type.instance));
+        ColumnSpecification cs10 = new ColumnSpecification("ks1", "cf1", new ColumnIdentifier("j", true), DynamicCompositeType.getInstance(ImmutableMap.of((byte)8, Int32Type.instance)));
+
+        ResultSet.ResultMetadata resultMetadata = new ResultSet.ResultMetadata(Arrays.asList(cs1, cs2, cs3, cs4, cs5, cs6, cs7, cs8, cs9, cs10));
         ResultSet rs = new ResultSet(resultMetadata, mock(List.class));
         Rows r1 = new Rows(rs);
         Rows r2 = overrideKeyspace(r1);
-        assertThat(r2.result.metadata.names.stream().map(cs -> cs.ksName)).containsExactly("ks1_123", "ks2_123");
+        assertThat(r2.result.metadata.names.stream().map(cs -> cs.ksName)).allMatch(k -> k.equals("ks1_123"));
```

Comment:
Noting the change to compute a new `resultMetadataId` in `ResultSet.withOverriddenKeyspace`, does it make sense to check that here with something like:
```
assertThat(r2.result.metadata.getResultMetadataId()).isNotSameAs(r1.result.metadata.getResultMetadataId());

```

Reply:
good idea